### PR TITLE
Apiservice

### PR DESCRIPTION
**Pull request by Celestimon**
Added facebook login/logout, app signup / delete account logic.

### CODE REVIEW

File: src/containers/FacebookHeader.js [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
     return {
         handleFacebookResponse: response => {
-            console.log(response)
-            dispatch(facebookLogin(response))
+            console.log(response);
+            request
+                .get("http://ec2-54-237-141-145.compute-1.amazonaws.com/v1/profile/")
+                // .set('X-CSRFToken', 'wQ1hh4q5DrIiriao8gfr5LiUwvoQig3LXxtcni8hg9jhTkoJmry0fGXgc2BiWGDG')
+                .set('Accept', 'text/html')
+                .auth('10203853660538839', 'EAAGfsEGo7q8BAKtdGmrnVZAmoUzHCD51DbPjZC2yDLxskEWJUaM25k5EJIZCneq5sJCyM0Lh3rtrvzWk6EMDZBRsiUpZAhH4dCg3YxiS6gGfMyeQEt01bCC2dUzzLv8FRcGOwEtyp56IPWupjwuDOLMyAj3wZBmHkGVQZA6y7NAu5SY3P7GHVtHuWn0jkriwUS6zW2uzmSGpgZDZD')
```
Comment (gautamgupta) [inline src/containers/FacebookHeader.js]: Is this yours?

Comment (Celestimon) [reply src/containers/FacebookHeader.js]: yes woops. will change to variables.

File: src/middleware/apiservice.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                            type: FACEBOOK_LOGOUT
+                        })
+                    }
+                })
+            break
+        case GAL_REMOVEACCOUNT:
+            console.log("middleware sending remove account request to backend");
+            console.log(action.userInfo)
+            request
+                .delete('http://ec2-54-237-141-145.compute-1.amazonaws.com/v1/profile/')
```
Comment (gautamgupta) [inline src/middleware/apiservice.js]: We can probably store `http://ec2-54-237-141-145.compute-1.amazonaws.com/v1` in our constants file.

File: src/reducers/accounts.js [hunk 3]
```diff
@@ -22,8 +24,26 @@ export const loggedIn = (initialState = false, action) => {
     switch (action.type) {
         case FACEBOOK_LOGIN:
             return Boolean(action.response.id)
+            break
```
Comment (gautamgupta) [inline src/reducers/accounts.js]: Don't need to break after return

File: src/containers/FacebookHeader.js [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
+                .end(function(err, res) {
+                    console.log(res.status)
+                    if (err && res.status === 403) {
+                        console.log("redirect to signup")
+                    }
+                })
+            dispatch(facebookLogin(response));
```
Comment (gautamgupta) [inline src/containers/FacebookHeader.js:27]: ES6 doesn't need semicolons after statements. Upto you to remove them though!

**Review by gautamgupta — COMMENTED** [review]
Looks good overall, added some comments.

File: src/actions/index.js [hunk 5]
```diff
@@ -4,11 +4,39 @@
 
 export const FACEBOOK_LOGIN = 'FACEBOOK_LOGIN'
 export const FACEBOOK_LOGOUT = 'FACEBOOK_LOGOUT'
+export const GAL_SIGNUP = 'GAL_SIGNUP'
```
Comment (gautamgupta) [inline src/actions/index.js:7]: What does GAL stand for?

**Pull request merged**